test/integration/change-streams/change_stream.test.ts

```diff
@@ -2699,4 +2699,47 @@ describe('ChangeStream resumability', function () {
       expect(changeStream.cursor.maxWireVersion).equal(maxWireVersion);
     }
   );
+
+  it(
+    'does not raise unhandled rejection errors',
+    { requires: { topology: '!single' } },
+    async function () {
+      if (globalThis.AbortSignal?.timeout == null) {
+        this.skipReason = 'test requires AbortSignal.timeout';
+        this.skip();
+      }
+
+      const unhandledRejections: AsyncIterableIterator<[reason: Error, promise: Promise<any>]> = on(
+        process,
+        'unhandledRejection',
+        { signal: AbortSignal.timeout(2000) }
+      );
+
+      changeStream = collection.watch();
+
+      const shouldErrorLoop = (async function () {
+        try {
```

Comment:
I don't fully follow the purpose of this test; in my understanding, this will return an error as long as it errors at any point in the iteration, but, given the context, aren't we expecting it to error immediately? i.e., on the first call to the iterator? if so, seems like we should insert a `break` in L2724, no?

Reply:
Yea break or return null, This test may not be worth keeping anymore with the global checker, since this problem we're trying to address is generic it just happened to be inside change streams